numpy/core/src/multiarray/iterators.c

```diff
@@ -1886,7 +1930,20 @@ static void neighiter_dealloc(PyArrayNeighborhoodIterObject* iter)
     Py_DECREF(iter->_internal_iter);
 
     array_iter_base_dealloc((PyArrayIterObject*)iter);
-    PyArray_free((PyArrayObject*)iter);
+    Py_TYPE(iter)->tp_free((PyObject *)iter);
+}
+
+static int
+neighiter_traverse(PyArrayNeighborhoodIterObject *iter, visitproc visit,
+        void *arg)
+{
+    if (iter->mode == NPY_NEIGHBORHOOD_ITER_CONSTANT_PADDING) {
+        if (PyArray_ISOBJECT(iter->_internal_iter->ao)) {
+            Py_VISIT(*(PyObject**)iter->constant);
+        }
+    }
+    Py_VISIT(iter->_internal_iter);
+    return 0;
```

Comment:
`neighiter_traverse` is never called

Reply:
I am considering to just undo the beighborhood iterator. It is an unused feature (within NumPy), which is meant to be used for C-side iteration of input arrays. (I bet scipy will be the only user).